### ISSUE TITLE
Hacked Vehicle list

### ISSUE THREAD

**Issue by s3kShUn61** [JoSchaap/GoT_Wasteland_V2.Stratis#60]
So I know the hackedVehicle check now shows the last owner of the vehicle, however it is very poorly optimized and the mission vehicles that spawn are also displayed in the hacked vehicles list. This means that a server that has been running for a couple hours now has a hacked vehicle list >20. I left my computer while loading the list for 10 mins and it still had not listed anything. Mainly can the mission vehicles be removed from being shown in the hacked vehicle list? It defeats the purpose since players can take mission vehicles and it looks like they are in hacked vehicles.

**Comment by AgentRev** [comment]
I'll look into that

**Comment by MercyfulFate** [comment]
The whole idea is kinda flawed in the respects that anyone one who knows what the are doing can remove the checksum from all vehicles and its useless. Or they can copy the checksum of another vehicle.

If I've read the code wrong then please correct me, otherwise maybe looking into a server side only solution maybe a better technique. Not sure of the performance impacts, but its harder to get around.

**Comment by AgentRev** [comment]
The checksums are stored server-side only, at least in the revision I'm working on.

Admins at TPG find the hacked vehicle list particularly useful, as it does help in banning many hackers. Don't forget that 99% of them are scriptkiddies with no actual coding skills whatsoever.

**Comment by Solo-** [comment]
I find the list quite usefull really, when you click on any vehicle and you see a normal player name driving it, then spectate that player untill it stops, teleport next to him and look inside the inventory of the vehicle, if you see a bunch of grenades it's hacked in and he's banned instantly, caught a cpl dozen players that way allready.

**Comment by Solo-** [comment]
What actually concerns me more is that players get into the server when it's really crowded and let it crash instantly with a screen telling me that the anti hack really sucks forcing me to stop/start the server.

**Comment by AgentRev** [comment]
Yeah I'm working on a new version of the anti-hack, which should stop the news banner thing.

Also Mercyful, I figured out how to prevent clients from overriding the checksum on all vehicles. Instead of assigning the checksum value to a "vehicleChecksum" variable, I could assign the checksum value itself as a variable, set to true. Example:

Old:
`_vehicle setVariable ["vehicleChecksum", call vChecksum, true];`

New:
`_vehicle setVariable [call vChecksum, true, false];`

The 3rd parameter is to whether or not broadcast the variable to clients. Since I moved everything related to hacked vehicles on the server, broadcasting is not needed anymore, and since `vChecksum` is a server-side key randomly generated on mission start, hackers won't be able to figure it out from their client.
